File: src/main/java/it/xpug/kata/birthday_greetings/MessageBuilder.java
package it.xpug.kata.birthday_greetings;

import org.jetbrains.annotations.NotNull;

import javax.mail.Message;
import javax.mail.MessagingException;
import javax.mail.Session;
import javax.mail.internet.InternetAddress;
import javax.mail.internet.MimeMessage;

public class MessageBuilder {

    private final Session session;

    public MessageBuilder(Session session) {
        this.session = session;
    }

    @NotNull
    Message buildMessage(Employee e) throws MessagingException {
        Message msg;
        msg = new MimeMessage(session);
        msg.setFrom(new InternetAddress("sender@here.com"));
        msg.setRecipient(Message.RecipientType.TO, new InternetAddress(e.getEmail()));
        msg.setSubject("Happy Birthday!");
        msg.setText("Happy Birthday, dear %NAME%!".replace("%NAME%",
                e.getFirstName()));
        return msg;
    }
}

File: src/main/kotlin/it/xpug/kata/birthday_greetings/MessageBuilder.kt
package it.xpug.kata.birthday_greetings

import javax.mail.Message
import javax.mail.Session
import javax.mail.internet.InternetAddress
import javax.mail.internet.MimeMessage

class MessageBuilder(private val session: Session) {

    fun buildMessage(e: Employee): Message {
        val msg: Message
        msg = MimeMessage(session)
        msg.setFrom(InternetAddress("sender@here.com"))
        msg.setRecipient(Message.RecipientType.TO, InternetAddress(e.email))
        msg.setSubject("Happy Birthday!")
        msg.setText("Happy Birthday, dear %NAME%!".replace("%NAME%",
                e.firstName))
        return msg
    }

}
